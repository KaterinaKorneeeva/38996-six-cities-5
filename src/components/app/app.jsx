import React from "react";
import PropTypes from "prop-types";
import {Switch, Route, BrowserRouter} from "react-router-dom";
import MainPage from "../main-page/main-page";
import AuthPage from "../auth-page/auth-page";
import FavoritesPage from "../favorites-page/favorites-page";
import OfferPage from "../offer-page/offer-page";
import {connect} from "react-redux";

const App = (props) => {
  const {offers, offer} = props;

  // console.log('props1111111111',props);
  // componentDidMount() {

  // };

  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/">
          <MainPage />
        </Route>
        <Route exact path="/login">
          <AuthPage />
        </Route>
        <Route exact path="/favorites">
          <FavoritesPage
            offers={offers}
          />
        </Route>
        <Route exact path="/offer/:id">
          <OfferPage
            offer = {offer}
            nearOffers={offers.slice(0, 3)}
          />
        </Route>
      </Switch>
    </BrowserRouter>
  );
};

App.propTypes = {
  offers: PropTypes.array.isRequired,
};

const mapStateToProps = ({DATA}) => ({
  // offers: DATA.selectedCity,
  // offerListByCity: DATA.offerListByCity
  offer: DATA.offer,
  offers: DATA.offerList,
  offerId: DATA.offerId,
});

export {App};
export default connect(mapStateToProps)(App);
