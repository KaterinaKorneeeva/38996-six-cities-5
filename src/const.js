export const CITY_LIST = [`Paris`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
export const SortingType = {
  POPULAR: `Popular`,
  LOW_TO_HIGH: `Price: low to high`,
  HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`
};

export const AppRoute = {
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  ROOT: `/`,
};

export const APIRoute = {
  HOTELS: `/hotels`,
  LOGIN: `/login`,
  COMMENTS: `/comments`,
  NEARBY: `/nearby`,
  OFFER: `/offer`,
};
