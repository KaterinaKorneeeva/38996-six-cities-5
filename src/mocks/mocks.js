export const offerList =
   [
     {
       city: {
         name: `Cologne`,
         location: {
           latitude: `50.938361`,
           longitude: `6.959974`,
           zoom: `13`
         }
       },
       previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
       images: [
         `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
         `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
         `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
       ],
       title: `The house among olive `,
       isFavorite: true,
       isPremium: false,
       rating: 3.8,
       type: `house`,
       bedrooms: 3,
       maxAdults: 4,
       price: 493,
       goods: [
         `Laptop friendly workspace`,
         `Washer`,
         `Towels`,
       ],
       host: {
         id: 25,
         name: `Angelina`,
         isPro: true,
         avatarUrl: `img/avatar-angelina.jpg`,
         avatar: `img/avatar-angelina.jpg`
       },
       description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
       location: {
         latitude: 50.957361,
         longitude: 6.9509739999999995,
         zoom: 16
       },
       id: 1
     }
   ];
export const comments =
  [
    {
      id: 1,
      user: {
        id: 11,
        isPro: false,
        name: `Jack`,
        avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`
      },
      rating: 5,
      comment: `This villa is perfect in every way: the view on mountains and waterfalls, the hot tub and the villa itself. The evening here became a great continuation of our journeys over country.`,
      date: `2020-10-10T13:38:44.753Z`
    },
    {
      id: 2,
      user: {
        id: 11,
        isPro: false,
        name: `Jack`,
        avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`
      },
      rating: 5,
      comment: `This villa is perfect in every way: the view on mountains and waterfalls, the hot tub and the villa itself. The evening here became a great continuation of our journeys over country.`,
      date: `2020-10-10T13:38:44.753Z`
    },
  ];
