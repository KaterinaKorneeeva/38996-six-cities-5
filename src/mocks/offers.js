import {getRandomInteger} from "../utils.js";

const PIC_URL = `https://picsum.photos/200/300?random=`;
const AVATAR_URL = `https://api.adorable.io/avatars`;

export default [
  {
    id: 0,
    rating: `5`,
    pictures: [
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
    ],
    isPremium: Boolean(1),
    title: `Wood and stone place`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    price: `60`,
    type: `room`,
    city: `Amsterdam`,
    coords: [52.3909553943508, 4.85309666406198],
    rooms: `4`,
    guests: `3`,
    features: [
      `wifi`,
      `heating`,
      `kitchen`,
      `cableTV`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`
    ],
    host: {
      name: `Simon`,
      status: `Pro`,
      photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
    },
    reviews: [
      {
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        author:
          {
            name: `Max`,
            photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
          },
        data: `2020-10-10`,
      },
      {
        text: `We booked at very short notice and Simon was quick to respond which was very helpful a long with details on how to check in. Accommodation is in a very good spot, within walking distance to tourist attractions like London eye etc. Apartment looked bigger in photos but was very cosy and had what we needed, would have loved if there was a tv even for some back ground noise. Over all good experience :)`,
        author:
          {
            name: `Annie`,
            photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
          },
        data: `2020-10-09`,
      }
    ]
  },
  {
    id: 1,
    rating: `5`,
    pictures: [
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
    ],
    isPremium: Boolean(0),
    title: `Beautiful & luxurious apartment at great location`,
    description: `The hostel serves a daily breakfast buffet and a set menu for dinner. There is also a bar in the hostel. There are many supermarkets in the surrounding streets, trendy bars, restaurants and shops in various squares in the Gr√†cia area, 15 minutes' walk away. The nearest Metro Station is Verdaguer, 500 meters from the hostel.`,
    price: `120`,
    type: `Apartment`,
    city: `Amsterdam`,
    coords: [52.369553943508, 4.85309666406198],
    rooms: `4`,
    guests: `4`,
    features: [`wifi`, `heating`, `kitchen`, `cableTV`],
    host: {
      name: `Angelina`,
      status: `Pro`,
      photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
    },
    reviews: [
      {
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        author:
          {
            name: `Max`,
            photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,

            rating: `4`,
          },
        data: `2020-10-10`,

      }
    ]
  },
  {
    id: 2,
    rating: `5`,
    pictures: [
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
    ],
    isPremium: Boolean(1),
    title: `Nice, cozy, warm big bed apartment`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    price: `180`,
    type: `house`,
    city: `Amsterdam`,
    coords: [52.3909553943508, 4.929309666406198],
    rooms: `4`,
    guests: `4`,
    features: [
      `wifi`,
      `heating`,
      `kitchen`,
      `cableTV`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`
    ],
    host: {
      name: `Angelina`,
      status: `Pro`,
      photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
    },
    reviews: [
      {
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        author:
          {
            name: `Max`,
            photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
            rating: `4`,
          },
        data: `2020-10-10`,

      }
    ]
  },
  {
    id: 3,
    rating: `5`,
    pictures: [
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
    ],
    isPremium: Boolean(0),
    title: `Nice, cozy, warm big bed apartment`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    price: `180`,
    type: `hotel`,
    city: `Brussels`,
    coords: [52.3809553943508, 4.939309666406198],
    rooms: `4`,
    guests: `4`,
    features: [`wifi`, `heating`, `kitchen`, `cableTV`],
    host: {
      name: `Angelina`,
      status: `Pro`,
      photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
    },
    reviews: [
      {
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        author:
          {
            name: `Max`,
            photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
            rating: `4`,
          },
        data: `April 2019`,

      }
    ]
  }
];
