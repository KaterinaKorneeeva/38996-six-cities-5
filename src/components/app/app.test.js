import React from "react";
import renderer from "react-test-renderer";
import {App} from "./app";
import {offerList} from "./../../mocks/mocks";

import {MemoryRouter} from "react-router-dom";
import configureStore from "redux-mock-store";
import {AuthorizationStatus} from "../../const";

import {Provider} from "react-redux";

const mockStore = configureStore()({
  USER: {
    authorizationStatus: AuthorizationStatus.AUTH,
    userInfo: {
      email: `email@mail.ru`
    }
  },
  DATA: {
    offerList,
    offerListByCity: offerList,
    selectedCity: `Amsterdam`,
    sortingType: `Popular`
  }
});
jest.mock(`../map/map`, () => `Map`);

it(`Should App render correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={mockStore}>
          <MemoryRouter>
            <App
              authorizationStatus = {AuthorizationStatus.AUTH}
              offerList = {offerList}
            />
          </MemoryRouter>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
