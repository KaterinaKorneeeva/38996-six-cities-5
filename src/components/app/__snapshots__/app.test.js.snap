// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should App render correctly 1`] = `
Array [
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <svg
      xlinkHref="http://www.w3.org/2000/svg"
    >
      <symbol
        id="icon-arrow-select"
        viewBox="0 0 7 4"
      >
        <path
          clipRule="evenodd"
          d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="icon-bookmark"
        viewBox="0 0 17 18"
      >
        <path
          d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
        />
      </symbol>
      <symbol
        id="icon-star"
        viewBox="0 0 13 12"
      >
        <path
          clipRule="evenodd"
          d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>,
  <div
    className="page page--gray page--main"
  >
    <header
      className="header"
    >
      <div
        className="container"
      >
        <div
          className="header__wrapper"
        >
          <div
            className="header__left"
          >
            <a
              className="header__logo-link header__logo-link--active"
            >
              <img
                alt="6 cities logo"
                className="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            className="header__nav"
          >
            <ul
              className="header__nav-list"
            >
              <li
                className="header__nav-item user"
              >
                <a
                  className="header__nav-link header__nav-link--profile"
                  href="#"
                  onClick={[Function]}
                >
                  <div
                    className="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    className="header__user-name user__name"
                  >
                    email@mail.ru
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      className="page__main page__main--index"
    >
      <h1
        className="visually-hidden"
      >
        Cities
      </h1>
      <div
        className="tabs"
      >
        <section
          className="locations container"
        >
          <ul
            className="locations__list tabs__list"
          >
            <li
              className="locations__item"
              onClick={[Function]}
            >
              <a
                className="locations__item-link tabs__item"
                href="#"
              >
                <span>
                  Paris
                </span>
              </a>
            </li>
            <li
              className="locations__item"
              onClick={[Function]}
            >
              <a
                className="locations__item-link tabs__item"
                href="#"
              >
                <span>
                  Brussels
                </span>
              </a>
            </li>
            <li
              className="locations__item"
              onClick={[Function]}
            >
              <a
                className="locations__item-link tabs__item tabs__item--active"
                href="#"
              >
                <span>
                  Amsterdam
                </span>
              </a>
            </li>
            <li
              className="locations__item"
              onClick={[Function]}
            >
              <a
                className="locations__item-link tabs__item"
                href="#"
              >
                <span>
                  Hamburg
                </span>
              </a>
            </li>
            <li
              className="locations__item"
              onClick={[Function]}
            >
              <a
                className="locations__item-link tabs__item"
                href="#"
              >
                <span>
                  Dusseldorf
                </span>
              </a>
            </li>
          </ul>
        </section>
      </div>
      <div
        className="cities"
      >
        <div
          className="cities__places-container container"
        >
          <section
            className="cities__places places"
          >
            <h2
              className="visually-hidden"
            >
              Places
            </h2>
            <b
              className="places__found"
            >
               
              1
               places to stay in 
              Amsterdam
            </b>
            <form
              action="#"
              className="places__sorting"
              method="get"
            >
              <span
                className="places__sorting-caption"
              >
                Sort by
              </span>
              <span
                className="places__sorting-type"
                onClick={[Function]}
                tabIndex="0"
              >
                Popular
                <svg
                  className="places__sorting-arrow"
                  height="4"
                  width="7"
                >
                  <use
                    xlinkHref="#icon-arrow-select"
                  />
                </svg>
              </span>
              <ul
                className="places__options places__options--custom "
              >
                <li
                  className="places__option places__option--active"
                  data-sorting="Popular"
                  onClick={[Function]}
                  tabIndex="0"
                >
                  Popular
                </li>
                <li
                  className="places__option "
                  data-sorting="Price: low to high"
                  onClick={[Function]}
                  tabIndex="0"
                >
                  Price: low to high
                </li>
                <li
                  className="places__option "
                  data-sorting="Price: high to low"
                  onClick={[Function]}
                  tabIndex="0"
                >
                  Price: high to low
                </li>
                <li
                  className="places__option "
                  data-sorting="Top rated first"
                  onClick={[Function]}
                  tabIndex="0"
                >
                  Top rated first
                </li>
              </ul>
            </form>
            <div
              className="cities__places-list tabs__content places__list"
            >
              <article
                className="cities__place-card place-card"
                onMouseOver={[Function]}
              >
                <div
                  className="cities__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/1"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="200"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"
                      width="260"
                    />
                  </a>
                </div>
                <div
                  className="place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        493
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button--active place-card__bookmark-button button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "76%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/1"
                      onClick={[Function]}
                    >
                      The house among olive 
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    house
                  </p>
                </div>
              </article>
            </div>
          </section>
          <div
            className="cities__right-section"
          >
            <Map
              cityCoord={
                Array [
                  52.38,
                  4.9,
                ]
              }
              offers={
                Array [
                  Object {
                    "bedrooms": 3,
                    "city": Object {
                      "location": Object {
                        "latitude": "50.938361",
                        "longitude": "6.959974",
                        "zoom": "13",
                      },
                      "name": "Cologne",
                    },
                    "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                    "goods": Array [
                      "Laptop friendly workspace",
                      "Washer",
                      "Towels",
                    ],
                    "host": Object {
                      "avatar": "img/avatar-angelina.jpg",
                      "avatarUrl": "img/avatar-angelina.jpg",
                      "id": 25,
                      "isPro": true,
                      "name": "Angelina",
                    },
                    "id": 1,
                    "images": Array [
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                    ],
                    "isFavorite": true,
                    "isPremium": false,
                    "location": Object {
                      "latitude": 50.957361,
                      "longitude": 6.9509739999999995,
                      "zoom": 16,
                    },
                    "maxAdults": 4,
                    "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                    "price": 493,
                    "rating": 3.8,
                    "title": "The house among olive ",
                    "type": "house",
                  },
                ]
              }
              type="cities_map"
            />
          </div>
        </div>
      </div>
    </main>
  </div>,
]
`;
